presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java

```diff
@@ -37,16 +45,70 @@
 {
-    private static final Principal PRINCIPAL = new TestingPrincipal("principal");
-    private static final Identity alice = new Identity("alice", Optional.of(PRINCIPAL));
-    private static final Identity bob = new Identity("bob", Optional.of(PRINCIPAL));
-    private static final Identity admin = new Identity("admin", Optional.of(PRINCIPAL));
-    private static final Identity nonAsciiUser = new Identity("\u0194\u0194\u0194", Optional.of(PRINCIPAL));
+    private static final Principal TESTING_PRINCIPAL = new TestingPrincipal("principal");
+    private static final Identity alice = new Identity("alice", Optional.of(TESTING_PRINCIPAL));
+    private static final Identity bob = new Identity("bob", Optional.of(TESTING_PRINCIPAL));
+    private static final Identity admin = new Identity("admin", Optional.of(TESTING_PRINCIPAL));
+    private static final Identity nonAsciiUser = new Identity("\u0194\u0194\u0194", Optional.of(TESTING_PRINCIPAL));
     private static final Set<String> allCatalogs = ImmutableSet.of("secret", "open-to-all", "all-allowed", "alice-catalog", "allowed-absent", "\u0200\u0200\u0200");
     private static final QualifiedObjectName aliceTable = new QualifiedObjectName("alice-catalog", "schema", "table");
     private static final QualifiedObjectName aliceView = new QualifiedObjectName("alice-catalog", "schema", "view");
     private static final CatalogSchemaName aliceSchema = new CatalogSchemaName("alice-catalog", "schema");
+    public static final String DEFAULT_CONFIG = "catalog.json";
     private TransactionManager transactionManager;
+    private Principal LDAP_PRINCIPAL;
+    private Principal KERBEROS_HOST_REALM;
+    private Principal KERBEROS_IP_REALM;
+    private Principal KERBEROS_REALM;
+    private Principal KERBEROS_HOST;
+    private Principal KERBEROS_SIMPLE;
+    private Properties oldSessionProperties;
+
+    @BeforeClass
+    public void setUp()
+    {
+        oldSessionProperties = setKerberosSessionProperties();
+        try {
+            LDAP_PRINCIPAL = new LdapPrincipal("CN=principal");
+        }
+        catch (InvalidNameException e) {
+            fail(e.toString());
+        }
+
+        KERBEROS_HOST_REALM = new KerberosPrincipal("principal/hostname@REALM");
+        KERBEROS_IP_REALM = new KerberosPrincipal("principal/192.1.1.1@REALM");
+        KERBEROS_REALM = new KerberosPrincipal("principal@REALM");
+        KERBEROS_HOST = new KerberosPrincipal("principal/hostname");
+        KERBEROS_SIMPLE = new KerberosPrincipal("principal");
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        for (Map.Entry<Object, Object> property : oldSessionProperties.entrySet()) {
+            System.setProperty((String) property.getKey(), (String) property.getValue());
+        }
+    }
+
+    private Properties setKerberosSessionProperties()
+    {
+        Properties properties = new Properties();
+        String kdcProperty = "java.security.krb5.kdc";
```

Comment:
I know this is gross, but you need either an /etc/krb5.conf file specified or these two properties if you're constructing a KerberosPrincipal. The other option is to subclass KerberosPrincipal, but then you're testing with a mock rather than a real principal. Thoughts?